api/app/Http/Controllers/BaseController.php

```diff
@@ -76,12 +73,17 @@ public function postOne(Request $request)
     public function putOne($id, Request $request)
     {
         $this->validator->with($request->all())->id($id)->validate();
-
-        return response($this->repository->createOrReplace($id, $request->all()), 201);
+        try{
+            $model = $this->repository->find($id);
+        }catch (ModelNotFoundException $e){
+            $model = $this->repository->getModel();
+        }
+        $model->fill($request->all());
+        return response((bool)$this->repository->save($model), 201);
```

Comment:
I'm ok with this pattern, I think it is a little clearer that having the repository handle it.